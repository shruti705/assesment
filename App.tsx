/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import HomeScreen from './src/component/HomeScreen/HomeScreen';
import QuestionnaireScreen from './src/component/Questionnaire/QuestionnaireScreen';
import ResultScreen from './src/component/Result/ResultScreen';
import {createStackNavigator} from '@react-navigation/stack';
import HeaderBackImage from './src/component/Stylecomponent/HeaderBackImage';

function App(): React.JSX.Element {
  const Stack = createStackNavigator();
  return (
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="Home"
        screenOptions={{
          headerBackImage: HeaderBackImage,
          headerStyle: {backgroundColor: '#1a2330'},
          headerTintColor: '#1C262E',
          headerTitleAlign: 'center',
        }}>
        <Stack.Screen
          name="Home"
          component={HomeScreen}
          options={{ headerShown: false }}
        />
        <Stack.Screen
          name="Questionnaire"
          component={QuestionnaireScreen}
          options={{
            title: 'Risk Profile Questionnaire',
            headerTintColor: '#FFFFFF',
          }}
        />
        <Stack.Screen
          name="Result"
          component={ResultScreen}
          options={{
            title: '',
          }}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

export default App;
