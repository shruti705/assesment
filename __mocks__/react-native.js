// __mocks__/react-native.js
const React = require('react');
const { View, Text } = require('react-native');

module.exports = {
  ...jest.requireActual('react-native'),
  View: jest.fn(() => <View />),
  Text: jest.fn(() => <Text />),
};

// __mocks__/@react-navigation/stack.js
const createStackNavigator = jest.fn(() => ({
  Navigator: ({ children }) => <>{children}</>,
  Screen: () => null,
}));

module.exports = { createStackNavigator };

// __mocks__/@react-navigation/native.js
const NavigationContainer = ({ children }) => <>{children}</>;

module.exports = { NavigationContainer };
