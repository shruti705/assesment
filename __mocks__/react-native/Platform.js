const Platform = {
    OS: 'ios', // or 'android', depending on your target platform
  };
  
  module.exports = Platform;
  