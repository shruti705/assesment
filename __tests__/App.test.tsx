import React from 'react';
import { render, cleanup, waitFor } from "@testing-library/react-native";
import { act } from "react-test-renderer";
import App from "../App"; // Ensure this path is correct
import renderer from 'react-test-renderer';
const MockNavigator = ({ children }: { children: React.ReactNode }) => <>{children}</>;
const MockScreen = ({ component: Component }: { name: string; component: React.ComponentType }) => (Component ? <Component /> : null);

jest.mock('@react-navigation/native', () => {
  const originalModule = jest.requireActual('@react-navigation/native');
  return {
    ...originalModule,
    useNavigation: () => ({
      navigate: jest.fn(),
    }),
    useRoute: () => ({
      params: {},
    }),
    NavigationContainer: ({ children }: { children: React.ReactNode }) => (
      <>{children}</>
    ),
  };
});

jest.mock('@react-navigation/stack', () => {
  return {
    createStackNavigator: () => {
      return { Navigator: MockNavigator, Screen: MockScreen };
    },
  };
});

jest.mock("../src/component/HomeScreen/HomeScreen", () => 'HomeScreen');
jest.mock("../src/component/Questionnaire/QuestionnaireScreen", () => 'QuestionnaireScreen');
jest.mock("../src/component/Result/ResultScreen", () => 'ResultScreen');

describe("AppNavigator", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    cleanup();
  });
  
  test('renders correctly', () => {
    const getByText = renderer.create(
      <App />
    ).toJSON();

    expect(getByText).toMatchSnapshot();
  });
});
