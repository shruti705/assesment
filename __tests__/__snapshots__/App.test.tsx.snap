// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          onGestureCancel={[Function]}
          pointerEvents="box-none"
          sheetAllowedDetents="large"
          sheetCornerRadius={-1}
          sheetExpandsWhenScrolledToEdge={true}
          sheetGrabberVisible={false}
          sheetLargestUndimmedDetent="all"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            collapsable={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              [
                {
                  "display": "flex",
                  "overflow": undefined,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              {
                "close": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  {
                    "flex": 1,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    [
                      {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      [
                        {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <View>
                        <Text>
                          HomeScreen
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`AppNavigator renders correctly 1`] = `
[
  <HomeScreen />,
  <QuestionnaireScreen />,
  <ResultScreen />,
]
`;
